### PR TITLE
RFC: Accept a Wrapping Token ID to unwrap and use a raw token

### PR DESCRIPTION
**Pull request by ewdurbin**
Our automation is not based on approle, but rather on https://github.com/hashicorp/vault-plugin-auth-kubernetes.

We are able to provide Wrapping Tokens with a short TTL that can be exchanged for a periodic token.

I think this also needs to short circuit approle config somehow when set

This is a WIP just to make explicit what I'm asking for. I'd love any feedback from @Caiyeon on what needs to happen to make this PR ready to merge.

### CODE REVIEW

**Comment by Caiyeon** [comment]
This is a good idea. I've been wanting to change nomad's bootstrap method (which is just a raw token file) into what you just described. A file with a wrapped token, no approle involved. This would fit both nomad and k8s deployments. 

The deployment for nomad environments would need to be changed, slightly. Breaking deployments means this change needs to go in a `v0.x.0` version, likely `v0.9.0`.

I've also been meaning to do a cleanup on the bootstrap process, and I won't ask you do to that in a PR. I'll merge this as-is soon, and do some rearrangements myself.

Tl;dr This is good as-is, and you can expect a full-fledged feature in `v0.9.0`
